src/appsettings.cpp

```diff
@@ -77,6 +83,10 @@ AppSettings::~AppSettings()
     m_settings.setValue(QStringLiteral("update/countryCode"), m_countryCode);
     m_settings.setValue(QStringLiteral("update/autoUpdate"), m_autoUpdate);
     m_settings.setValue(QStringLiteral("update/autoUpdateTime"), m_autoUpdateTime);
+    m_settings.setValue(QStringLiteral("application/notifyLevel"), m_notifyLevel);
+    QVariant notifyIds;
+    notifyIds.setValue(m_notifyIds);
+    m_settings.setValue(QStringLiteral("application/notifyIds"), notifyIds);
```

Comment:
Why not directly cast it to `QVariant`?

Reply:
It would be nice I agree, but it won't compile like that unfortunately, unless there's a trick I'm missing? I guess it's because `setValue()` is [templated](https://git.sailfishos.org/mer-core/qtbase/blob/mer-5.6/src/corelib/kernel/qvariant.h#L341), whereas the constructor is just a [bunch of overrides](https://git.sailfishos.org/mer-core/qtbase/blob/mer-5.6/src/corelib/kernel/qvariant.h#L193) for known conversions (of which `QList<int>` isn't one).

Reply:
Would `m_settings.setValue(QStringLiteral("application/notifyIds"), QVariant(m_notifyIds))`  work?

Reply:
The same issue arises unfortunately (it won't compile like that).

Reply:
Ok, then I give up :)

Reply:
It was a good idea to suggest it. I'd prefer to avoid the current mess, but this is the tidiest I could find.